brian2/tests/test_cpp_standalone.py

```diff
@@ -59,6 +59,35 @@ def test_multiple_connects():
     assert len(S) == 2 and len(S.w[:]) == 2
     reset_device()
 
+@pytest.mark.cpp_standalone
+@pytest.mark.standalone_only
+def test_device_cache_synapses():
+    # Check that we can ask for known synaptic information at runtime
+    set_device('cpp_standalone', build_on_run=False)
+    G = NeuronGroup(10, 'v:1')
+    S = Synapses(G, G, 'w:1')
+    S.connect(i=[0], j=[0])
+    assert len(S) == 1
+    assert_equal(S.i[:], [0])
+    assert_equal(S.j[:], [0])
+    
+    S.connect(i=[1], j=[1])
+    assert len(S) == 2
+    assert_equal(S.i[:], [0, 1])
+    assert_equal(S.j[:], [0, 1])
+
+    S.connect(p=0.1)  # We can't know anything about synapses anymore
+
+    with pytest.raises(NotImplementedError):
+        len(S)
+    
+    with pytest.raises(NotImplementedError):
+        S.i[:]
+
+    with pytest.raises(NotImplementedError):
+        S.j[:]
+
```

Comment:
It might be worth having another test setting the indices to concrete values again to check that they get updated properly after using something like the `p=0.1` construct.

Reply:
Done.

Reply:
👍